### DISCUSSION TITLE
Is there a way to "un-nest" association?

### DISCUSSION THREAD

**Discussion by oskargargas** [okuramasafumi/alba#290] [Q&A] [answered]
Hi!

Let's say I have following models:
- Foo
- Bar
- FooBar

`Foo` has many  `FooBar`.
`Foo` has many  `Bar` through `FooBar`.

`FooBar` also has additional field, let's say `A`.

Situation:
I have access to `FooBar` instance but the API is not interested in `FooBar`. API wants to get `Bar` and field `A` from `FooBar`.

In JBuilder I used to do it this way:
```ruby
json.partial! "bar/bar", document: foo_bar.bar
json.field_a foo_bar.a
```
It would put everything at the same nesting level. Something like:
```json
{
   // from Bar model via partial
  "id": 1,
  "name": "Bar",

   // from FooBar model
  "field_a": "A"
}
```

Is something like that possible in Alba? In best case scenario I'd love to write it like this. Though I can't find anything alike in docs or examples.
```ruby
class FooBarResource
  include Alba::Resource

  one :bar, resource: BarResource, unnest: true
  attributes :a
end
```

**Comment by trevorturk** [comment, marked as answer]
There might be a better answer for this, but one possible solution would be to simply define a method in your resource which calls the relevant method on the associated model.

**Reply by oskargargas** [reply]
Not sure how it could work. Could you explain a bit more?

I currently use something like that to access `FooBar.a` from `Bar` using `delegate`. But I don't see how I could access all fields of `Bar` from `FooBar` without delegating all fields from `Bar` to `FooBar`.

**Reply by trevorturk** [reply]
Yeah, that's what I mean is to delegate the other methods you need. But it would be annoying if you have a lot of them. You can define methods or use delegation in an Alba resource, though, I believe. So perhaps it's doable without too much hassle.
